core/src/com/unciv/ui/worldscreen/WorldScreen.kt

```diff
@@ -573,7 +574,10 @@ class WorldScreen(val viewingCiv:CivilizationInfo) : CameraStageBaseScreen() {
         // Also, the reaction of other popups like 'disband this unit' to back/esc feels nicer this way.
         // After removeListener just in case this is slow (enumerating all stage actors)
         if (hasOpenPopups()) {
-            closeAllPopups()
+            val closedName = closeOneVisiblePopup() ?: return
+            if (closedName.startsWith(Constants.tutorialPopupNamePrefix)) {
+                tutorialController.removeTutorial(closedName.substring(Constants.tutorialPopupNamePrefix.length))
```

Comment:
Please use closedName.removePrefix(Constants.tutorialPopupNamePrefix),
behind the hood it does exactly the same but it's more immediately obvious what we're doing

Reply:
Yes looks much nicer - but under the hood, unless the compiler really catches the relation to the preceding startsWith, that's a second comparison as far as I understand removePrefix. But - impact negligible.